File: backend/core-api/models/deployment.go
package models

import "time"

type Deployment struct {
	ID        string    `json:"id"`
	ClusterID string    `json:"cluster_id"`
	Version   string    `json:"version"`
	Strategy  string    `json:"strategy"`       // canary|blue-green|rolling
	Target    int       `json:"target_percent"` // for canary
	Status    string    `json:"status"`         // pending|in-progress|rolled_out|rolled_back|failed
	StartedAt time.Time `json:"started_at"`
	UpdatedAt time.Time `json:"updated_at"`
	Logs      []string  `json:"logs"`
}

type StartDeploymentRequest struct {
	ClusterID     string `json:"cluster_id"`
	Version       string `json:"version"`
	Strategy      string `json:"strategy"`
	TargetPercent int    `json:"target_percent"`
}

File: frontend/src/models/deployment.ts
export interface Deployment {
  id: string;
  cluster_id: string;
  version: string;
  strategy: string;
  target_percent?: number;
  status: string;
  started_at?: string;
  updated_at?: string;
  logs?: string[];
}

export interface StartDeploymentRequest {
  cluster_id: string;
  version: string;
  strategy: string;
  target_percent?: number;
}
